src/Operations.jl

```diff
@@ -587,18 +582,25 @@ function collect_artifacts(pkg_root::String; platform::AbstractPlatform=HostPlat
     return artifacts_tomls
 end
 
-
-function download_artifacts(env::EnvCache, pkgs::Vector{PackageSpec};
+function download_artifacts(env::EnvCache;
                             platform::AbstractPlatform=HostPlatform(),
                             julia_version = VERSION,
                             verbose::Bool=false,
                             io::IO=DEFAULT_IO[])
-    for pkg in pkgs
+    pkg_roots = String[]
+    for (uuid, pkg) in env.manifest
+        pkg = manifest_info(env.manifest, uuid)
         pkg_root = source_path(env.project_file, pkg, julia_version)
-        pkg_root === nothing && continue
+        pkg_root === nothing || push!(pkg_roots, pkg_root)
+    end
+    envpkg = env.pkg
+    envpkg === nothing || push!(pkg_roots, envpkg.path)
+    for pkg_root in pkg_roots
         for (artifacts_toml, artifacts) in collect_artifacts(pkg_root; platform)
             # For each Artifacts.toml, install each artifact we've collected from it
             for name in keys(artifacts)
+                @show name, artifacts[name], artifacts_toml
```

Comment:
Just to make sure you don't accidentally leave the `@show` in here

Reply:
Jeez, today has really been a spectacle in this type of stuff for me. :sweat: